core/src/main/scala/kafka/log/Log.scala

```diff
@@ -387,7 +387,18 @@ class Log(@volatile var dir: File,
         fileSuffix = SwapFileSuffix)
       info(s"Found log file ${swapFile.getPath} from interrupted swap operation, repairing.")
       recoverSegment(swapSegment)
-      val oldSegments = logSegments(swapSegment.baseOffset, swapSegment.readNextOffset)
+
+      // Assigned to a new value
+      val nextOffset = swapSegment.readNextOffset
+      var oldSegments = logSegments(swapSegment.baseOffset, nextOffset)
+
+      // This means swap file is empty. Because of that baseOffset and nextOffset is the same.
+      // logSegments method above would return empty list. Here we are making sure that
+      // the segment with the baseOffset is returned as well.
+      if(oldSegments.isEmpty && swapSegment.baseOffset == nextOffset) {
```

Comment:
@kondetibharat , even the oldSegments is not empty, you still need to add the empty swap file into the oldSegments, or you may end up with oldSegments is not empty but the empty swap file is still missing.

Reply:
This will not happen in logSegments method. If the oldSegments list is not empty, then it will definitely contain the segment for the swap file. So we should be fine